Book.cpp

```diff
@@ -311,3 +311,167 @@ void Book::getAllBooks() {
+	}
+	
+	char* getEmployeeID(){
+		return employeeID;
+	}
+
+	char* getMemberID(){
+		return membershipID;
+	}
+
+	booksIssued* getBooksIssued(){
+		return Books;
+	}
+
+	// void alterEmployeeID(){
+	// 	cin.get(employeeID,10);
+	// }
+
+	// void alterMembershipID(){
+	// 	cin.get(membershipID,10);
+	// }
+
+	void addEmployee(Employee& obj){
+		// char* temp = getMemberID();
+		// cin.get(obj.employeeID,10);
+		cin>>obj.employeeID;
+		// cin.get(obj.membershipID,10);
+		cin>>obj.membershipID;
+		if(existingEmployee() == false){		//no existing ID of the employee
+			f.open("EmployeeData.dat",ios::out|ios::app|ios::binary);
+			f.write((char*)&obj,sizeof(obj));
+			f.close();
+			cout<<"Employee Added"<<endl;
+		}
+		else{
+			cout<<"Already Registered."<<endl;
+		}
+	}
+
+	void deleteEmployee(Employee& obj){
+		fstream fin;
+		fstream fout;
+		Employee temp;
+		fin.open("temp.dat",ios::in|ios::out|ios::binary);
+		fout.open("EmployeeData.dat",ios::in|ios::out|ios::app|ios::binary);
+		fin.seekg(0,ios::beg);
+		while(fin.read((char*)&temp,sizeof(temp))){
+			if(strcmpi(obj.getMemberID(),temp.getMemberID()) == 0){
+				cout<<obj.employeeID<<" "<<temp.employeeID;
+				continue;
+			}
+			else{
+				fin.write((char*)&temp,sizeof(temp));
+			}
+		}
+		fin.close();
+		fout.close();
+		remove("EmployeeData.dat");
+		rename("temp.dat","EmployeeData.dat");
+
+	}
+
+	void addBook(Employee& b){
+		int nBooks;
+		cout<<"Number of books being issued: ";
+		cin>>nBooks;
+		while(nBooks--){
+			cout<<"1"<<endl;
+		++bookCount;
+		int idx = bookCount - 1;
+		cout<<"idx: "<<idx<<endl;
+		cin>>b.Books[idx].LibraryID;
+		cin>>b.Books[idx].ISBN;
+		}
+		fstream ftemp;
+		ftemp.open("temp.dat", ios::in | ios::out | ios::binary | ios::app);
+		f.open("EmployeeData.dat", ios::in | ios::binary);
+		Employee temp;
+		while(f.read((char*)&temp,sizeof(temp))){
+
+			if(strcmpi(b.getMemberID(),temp.getMemberID()) == 0){
+				ftemp.write((char*)&b,sizeof(b));
+				continue;
+			}
+			ftemp.write((char*)&temp,sizeof(temp));
+		}
+		f.close();
+		ftemp.close();
+		cout<<"Book added"<<endl;
+		remove("EmployeeData.dat");
+		rename("temp.dat","EmployeeData.dat");
+	}
+
+	bool existingEmployee(){
+		Employee temp;
+		f.open("EmployeeData.dat", ios::in | ios::binary);
+		while(f.read((char*)&temp,sizeof(temp))){
+				if(strcmpi(temp.employeeID,getEmployeeID()) == 0){
+					f.close();
+					return true;
+				}
+		}
+		f.close();
+		return false;
+
+	}
+
+	void showInfo(Employee& obj){
+		cout<<"EmployeeID: "<<obj.employeeID<<endl;
+		cout<<"MembershipID: "<<obj.membershipID<<endl;
+	}
+
+	void showInformation(){
+		f.open("EmployeeData.dat", ios::in | ios::binary);
+		Employee temp;
+		while(f.read((char*)&temp,sizeof(temp))){
+			showInfo(temp);
+		}
+	}
+
+	
+};
+
+int main(){
+	Employee e1,e2;
+	// e1.addEmployee(e1);
+	// e1.deleteEmployee(e1);
+	e1.addEmployee(e1);
+	e2.addEmployee(e2);
+	e2.deleteEmployee(e1);
+	cout<<"list is"<<endl;
+	e1.showInformation();
+
+}

```

Comment:
So we need a function to display the books the person has issued and then also the fine calculation.
I think that should be the end of the review, if I notice that you have missed something, I'll notify you.